cairo-contracts/packages/apps/src/transfer/components/transfer.cairo

```diff
@@ -283,19 +283,35 @@ pub mod TokenTransferComponent {
 
     #[embeddable_as(CreateIbcToken)]
     impl CreateTokenImpl<
-        TContractState, +HasComponent<TContractState>, +Drop<TContractState>,
+        TContractState,
+        +HasComponent<TContractState>,
+        +Drop<TContractState>,
+        impl Ownable: OwnableComponent::HasComponent<TContractState>,
     > of ICreateIbcToken<ComponentState<TContractState>> {
         fn create_ibc_token(
-            ref self: ComponentState<TContractState>,
-            chan_id_on_b: ChannelId,
-            base_denom: ByteArray,
+            ref self: ComponentState<TContractState>, denom: PrefixedDenom,
         ) -> ContractAddress {
-            let token = self.create_token(base_denom.clone(), 0);
-            let trace_prefix = TracePrefixTrait::new(TRANSFER_PORT_ID(), chan_id_on_b);
-            let prefixed_denom = PrefixedDenom {
-                trace_path: array![trace_prefix], base: Denom::Hosted(base_denom),
-            };
-            self.record_ibc_token(prefixed_denom, token.address);
+            let mut token = self.get_token(denom.key());
+            assert(token.is_zero(), TransferErrors::TOKEN_ALREADY_EXISTS);
```

Comment:
Not sure if Soares wanted to throw an error here. He wanted something like `get_or_create_ibc_token`.

Reply:
I think the `ibc_token_address` API already serves as the `getter` if needed, though with this difference, it offloads the key derivation process from the `PrefixedDenom` off-chain for a bit of cost reduction.